test/functional/apps/context/_discover_navigation.js

```diff
@@ -35,7 +35,10 @@ export default function ({ getService, getPageObjects }) {
   describe('context link in discover', () => {
     before(async () => {
       await PageObjects.timePicker.setDefaultAbsoluteRangeViaUiSettings();
-      await kibanaServer.uiSettings.update({ 'doc_table:legacy': true });
+      await kibanaServer.uiSettings.update({
+        'doc_table:legacy': true,
+        defaultIndex: 'logstash-*',
```

Comment:
without this change , `logstash*` would be selected, and the test would fail

Reply:
@kertal I took lots of time, but it seems I found the reason. In https://github.com/elastic/kibana/pull/89264 you added 
https://github.com/elastic/kibana/blob/8d17ce0719ca487d519a94bef603b54a049fff90/test/functional/apps/context/_discover_navigation.js#L53-L55

which removes a value set in `index.js` file https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/test/functional/apps/context/index.js#L22

So, `defaultIndex` setting is `undefined`. Data plugin needs it, so it loads all the known `index-pattern`s and set the very first (which is `logstash*`) as `defaultPattern` https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts#L24

Why does it not fail on the master branch? All the SO docs have the same score. In this case, Elasticsearch sort by internal document Id (according to https://discuss.elastic.co/t/default-sort-for-same-score-documents/15107). 
In the current PR, Kibana migrates documents, which might cause internal document Id regeneration.
@mattkime Is it something that might affect the end-users? 
What do you want to achieve by using the head element of the `pattern` array in https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts#L40?
Do you need *any* pattern? The eldest known pattern? The latest known pattern? 
This comment doesn't explain much https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts#L38

Reply:
@mattkime confirmed it should return any `index-pattern`. Then we are good to go with the current solution.

Reply:
@mshustov great detective work! 🕵️